src/main/java/com/impactupgrade/nucleus/controller/MailchimpController.java

```diff
@@ -89,82 +74,83 @@ public Response messageEvent(
   }
 
   // Mailchimp sends a GET to make sure the webhook is available (silly...)
-  @Path("/webhook/message")
+  @Path("/webhook/audience")
   @GET
   public Response messageEvent() throws Exception {
     return Response.status(200).build();
   }
   
-  private void processEvent(Event event, Environment env) throws Exception {
-    if (event == null) {
+  private void processCampaignEvent(AudienceEvent event, Environment env) throws Exception {
+    if (!"sent".equalsIgnoreCase(event.status)) {
       return;
     }
-    if ("send".equalsIgnoreCase(event.eventType)) {
-      // Using sender::recipient::sent-date
-      // as a conversation id 
-      Date sentAt = Date.from(Instant.ofEpochSecond(event.message.timestamp));
-      String conversationId = event.message.sender + "::" + event.message.email + "::" + new SimpleDateFormat(DATE_FORMAT).format(sentAt);
-      env.activityService().upsertActivityFromEmail(
-          event.message.email,
-          EMAIL,
-          conversationId,
-          event.message.subject); // using subject instead of message body (body n\a in the webhook's payload)
-    } else {
-      env.logJobInfo("skipping event type {}...", event.eventType);
-    }
-  }
-  
-  public static final class MessageWebhookPayload {
-    @JsonProperty("mandrill_events")
-    public List<Event> events;
-  }
 
-  @JsonIgnoreProperties(ignoreUnknown = true)
-  public static final class Event {
-    @JsonProperty("_id")
-    public String id;
-    @JsonProperty("event")
-    public String eventType;
-    @JsonProperty("msg")
-    public Message message;
-
-    //@JsonProperty("ts")
-    //public Long timestamp;
-    //public String url;
-    //public String ip;
-    //@JsonProperty("user_agent")
-    //public String userAgent;
-    //public Object location;
-    //@JsonProperty("user_agent_parsed")
-    //public List<Object> userAgentParsed;
-    
+    // TODO: Clunky way to retrieve the specific account keys, when all we have is the List ID..
+    EnvironmentConfig.Mailchimp mailchimpConfig = null;
+    for (EnvironmentConfig.Mailchimp _mailchimpConfig : env.getConfig().mailchimp) {
+      for (EnvironmentConfig.CommunicationList communicationList : _mailchimpConfig.lists) {
+        if (event.listId.equalsIgnoreCase(communicationList.id)) {
+          mailchimpConfig = _mailchimpConfig;
+          break;
+        }
+      }
+    }
+    if (mailchimpConfig == null) {
+      env.logJobError("unable to find ListID={}", event.listId);
+      return;
+    }
+    MailchimpClient mailchimpClient = new MailchimpClient(mailchimpConfig, env);
+
+    List<SentToInfo> sentTos = mailchimpClient.getCampaignRecipients(event.id);
+    List<String> emails = sentTos.stream()
+        .filter(member -> member.status == SentToInfo.Status.SEND)
+        .map(member -> member.email_address)
+        .toList();
+
+    ContentInfo contentInfo = mailchimpClient.getCampaignContent(event.id);
```

Comment:
@VSydor This is 404'ing for me. Could you experiment with the API? The Kotlin code in the client lib looks correct, compared to the MC doc: https://mailchimp.com/developer/marketing/api/campaign-content/get-campaign-content/

Reply:
Will do!